packages/table/src/table.tsx

```diff
@@ -356,7 +356,20 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
         newRowHeights = Utils.arrayOfLength(newRowHeights, numRows, defaultRowHeight);
         newRowHeights = Utils.assignSparseValues(newRowHeights, rowHeights);
 
-        const newSelectedRegions = (selectedRegions == null) ? this.state.selectedRegions : selectedRegions;
+        const numCols = newColumnWidths.length;
+
+        let newSelectedRegions = selectedRegions;
+        if (selectedRegions == null) {
```

Comment:
yeah that too. we should have a `warn` utility function that does that check for us though